src/backends/redis.ts

```diff
@@ -98,14 +98,23 @@ export default class RedisBackend implements CeleryBackend {
   ): Promise<["OK", number]> {
     return this.set(
       `${keyPrefix}${taskId}`,
-      JSON.stringify({
-        status: state,
-        result,
-        traceback: null,
-        children: [],
-        task_id: taskId,
-        date_done: new Date().toISOString()
-      })
+      state == 'FAILURE' ?
+        JSON.stringify({
+          status: state,
+          result: null,
+          traceback: result,
+          children: [],
+          task_id: taskId,
+          date_done: new Date().toISOString()
+        }) : 
+        JSON.stringify({
+          status: state,
+          result,
+          traceback: null,
+          children: [],
+          task_id: taskId,
+          date_done: new Date().toISOString()
+        })
```

Comment:
What if write conditional on just result?
```suggestion
      state == 'FAILURE' ?
        JSON.stringify({
          status: state,
          result: state == 'FAILURE' ? null : state,
          traceback: result,
          children: [],
          task_id: taskId,
          date_done: new Date().toISOString()
        })
```